wgpu-core/src/device/mod.rs

```diff
@@ -2754,29 +2759,24 @@ impl<A: HalApi> Device<A> {
         };
 
         if validated_stages.contains(wgt::ShaderStages::FRAGMENT) {
-            for (i, state) in color_targets.iter().enumerate() {
-                match io.get(&(i as wgt::ShaderLocation)) {
-                    Some(output) => {
-                        validation::check_texture_format(state.format, &output.ty).map_err(
-                            |pipeline| {
-                                pipeline::CreateRenderPipelineError::ColorState(
-                                    i as u8,
-                                    pipeline::ColorStateError::IncompatibleFormat {
-                                        pipeline,
-                                        shader: output.ty,
-                                    },
-                                )
+            for (i, output) in io.iter() {
+                if let Some(state) = color_targets.get(*i as usize) {
+                    let format = if let Some(st) = state.as_ref() {
+                        st.format
+                    } else {
+                        return Err(
+                            pipeline::CreateRenderPipelineError::InvalidFragmentOutputLocation(*i),
+                        );
+                    };
+                    validation::check_texture_format(format, &output.ty).map_err(|pipeline| {
+                        pipeline::CreateRenderPipelineError::ColorState(
+                            *i as u8,
+                            pipeline::ColorStateError::IncompatibleFormat {
+                                pipeline,
+                                shader: output.ty,
                             },
-                        )?;
-                    }
-                    None if state.write_mask.is_empty() => {}
```

Comment:
I don't quite understand what this code is meant to do, but, this test seems to have disappeared. Why is that?

Reply:
Since I changed the order of the loop and used shader output `@location` index to fetch the color_target, ` None if state.write_mask.is_empty()` is no longer needed.